sql/mob_spawn_mods.sql

```diff
@@ -47,48 +47,224 @@ INSERT INTO `mob_spawn_mods` VALUES (16867642,55,180,1); -- IDLE_DESPAWN: 180
+INSERT INTO `mob_spawn_mods` VALUES (16912661,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912682,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912683,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912684,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912685,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912687,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912688,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912689,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912690,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912691,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912692,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912697,71,2,1); -- LEADER: 2
+INSERT INTO `mob_spawn_mods` VALUES (16912698,71,-1,1); -- LEADER: -1
+INSERT INTO `mob_spawn_mods` VALUES (16912699,71,-2,1); -- LEADER: -2
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912700,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912701,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912707,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912708,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912775,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912776,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912780,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912781,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912785,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912786,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912787,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912788,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912789,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912790,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912792,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912793,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912794,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912795,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912796,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912797,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912800,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912801,71,-1,1); -- LEADER: -1
+
+-- Om'Xzomit
+INSERT INTO `mob_spawn_mods` VALUES (16912805,71,1,1); -- LEADER: 1
+INSERT INTO `mob_spawn_mods` VALUES (16912806,71,-1,1); -- LEADER: -1
```

Comment:
Couldnt this be placed in mob scripts instead of the database?

There is awork going on to retire this table.

Reply:
The only way to do this would be if the mother and children have different scripts. Currently they have the same script and so there is no way to distinguish them from inside the script. This database allows us to individualize them.

Reply:
If we add this into the `ID.lua`:
`ULXZOMIT_OFFSET = GetTableOfIDs('Ulxzomit'),`

Inside the Ulxzomit script we can put something like:

```lua
local leaderTable =
{
    ID.mob.ULXZOMIT_OFFSET[1],
    ID.mob.ULXZOMIT_OFFSET[4],
    ID.mob.ULXZOMIT_OFFSET[7],
    ID.mob.ULXZOMIT_OFFSET[10],
    ID.mob.ULXZOMIT_OFFSET[13],
    ID.mob.ULXZOMIT_OFFSET[16],
    ID.mob.ULXZOMIT_OFFSET[19],
    ID.mob.ULXZOMIT_OFFSET[22],
    ID.mob.ULXZOMIT_OFFSET[25],
    ID.mob.ULXZOMIT_OFFSET[28],
    ID.mob.ULXZOMIT_OFFSET[31],
    ID.mob.ULXZOMIT_OFFSET[34],
    ID.mob.ULXZOMIT_OFFSET[37],
    ID.mob.ULXZOMIT_OFFSET[40],
    ID.mob.ULXZOMIT_OFFSET[43],
}

entity.onMobInitialize = function(mob)
    local mobId    = mob:getID()
    local leaderId = leaderTable[mobId]

    if leaderId ~= nil then
        mob:setMobMod(xi.mobMod.LEADER, 1)

        local childOne = mob:getID + 1
        local childTwo = mob:getID +2
        childOne:setMobMod(xi.mobMod.LEADER, -1)
        childTwo:setMobMod(xi.mobMod.LEADER, -2)
    end
end
```

Reply:
Just a brief glance, but even that could be simplified with some modulo action on the tabled result

Reply:
That's a good idea @Frankie-hz! I can change things to work like that. A modulo might work for the Ul'Xzomit since I think they all have 2 babies each but the Om'Xzomit have 1 or 2 babies each.

Reply:
I don't really see a pattern for Omxzomit so would have to think of something else for those.

Reply:
Frank's suggestion mostly worked. I have every mob calculate its own LEADER value and put the logic in `xi.follow.assignLeaderMod()` for easy reuse.